frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -425,7 +425,12 @@ function DatasetEditor(props) {
             onChange={onChangeEditorTab}
             options={[
               { id: "query", name: t`Query`, icon: "notebook" },
-              { id: "metadata", name: t`Metadata`, icon: "label" },
+              {
+                id: "metadata",
+                name: t`Metadata`,
+                icon: "label",
+                disabled: !result?.data,
```

Comment:
@daltojohnso @kulyk should we maybe use an existing selector here instead? e.g. `getResultsMetadata` is not null or something like that

Reply:
Ya might as well reuse the `getResultsMetadata` selector here as it essentially does what this line does already + checks for the presence of `result.data.results_metadata`.